### ISSUE TITLE
PATTERNS do not match any on InactiveRecipientError

### ISSUE THREAD

**Issue by travisjhilton** [ActiveCampaign/postmark-gem#128]
https://github.com/ActiveCampaign/postmark-gem/blob/f411c3474391210c43ae1d9ce5ea6c1c0198b609/lib/postmark/error.rb#L78

The message that is returned for an InactiveRecipientError does not match any of the regex in PATTERNS. The message that is returned:
`{"ErrorCode"=>406, "Message"=>"You tried to send to recipient(s) that have been marked as inactive. Found inactive addresses: HardBounce@bounce-testing.postmarkapp.com. Inactive recipients are ones that have generated a hard bounce, a spam complaint, or a manual suppression."}`

**Comment by obie** [comment]
This appears to be broken again. Perhaps the error message from the API changed?

**Comment by obie** [comment]
@balvig heads up

**Comment by balvig** [comment]
Thanks for pointing this out @obie! 🙏 
(cc @pgraham3, hope all is well 🙂)

**Comment by pgraham3** [comment]
@obie Can you please let me know what behaviour you are seeing vs what you would expect? Are you still seeing that this issue has resurfaced?

Testing `v1.25.1`of this gem I am seeing `Postmark::InactiveRecipientError` errors being raised successfully and the inactive address(es) available in the error's `recipients`, as expected:

```ruby
server_token = '...'

client = Postmark::ApiClient.new(server_token, http_open_timeout: 15)

begin
  client.deliver(
    from: 'sender@domain.com',
    to: 'test-email-1@example.com',
    subject: 'Test Email',
    text_body: 'This is a test email'
  )
rescue Postmark::InactiveRecipientError => e
  puts e.message
  puts "Recipients: #{e.recipients.inspect}"
end

```

Outputs:

> You tried to send to recipient(s) that have been marked as inactive. Found inactive addresses: test-email-1@example.com. Inactive recipients are ones that have generated a hard bounce, a spam complaint, or a manual suppression.
Recipients: ["test-email-1@example.com"]

Thanks!

**Comment by mkwiatkowski** [comment]
This bug resurfaced once more.

Using @pgraham3 's snippet above I get the following output:

```
You tried to send to recipient(s) that have been marked as inactive. Found inactive addresses: . Inactive recipients are ones that have generated a hard bounce, a spam complaint, or a manual suppression.
Recipients: []
```

**Comment by pgraham3** [comment]
@mkwiatkowski I cannot reproduce this quite yet... I am seeing the expected result still. 

What do you have for the output of the following:

`bundle info postmark-rails`

`bundle info postmark`

Can you tell me more about your test setup? I.e. one inactive address, multiple addresses, etc...

**Comment by mkwiatkowski** [comment]
@pgraham3 That's weird, I don't think it's app-dependant. It's as if something changed on the Postmark servers' side. Here's a minimal POC: https://github.com/mkwiatkowski/postmark-issue-128-poc Just clone the repo, bundle install and run with `ruby app.rb API_KEY`.

**Comment by pgraham3** [comment]
@mkwiatkowski Thanks for the example! I just used it with an inactive address in the `To` with my own token and see the expected output:

<img width="1801" alt="Screenshot 2025-01-15 at 11 07 45 AM" src="https://github.com/user-attachments/assets/c1c34a70-8d2e-4777-a5e0-15489c6c7f0d" />

Note the inactive recipient I tested with was inactive due to a Hard Bounce, prior to testing this. Is your inactive recipient address you are testing with also a Hard Bounce or perhaps some other type? Just trying to figure out why you are seeing different results. Does the email address have anything special for the local part of the email address, like someone**+**something@somewhere.com ?

**Comment by mkwiatkowski** [comment]
@pgraham3 I tested the address you've used in your test code, i.e. test-email-1@example.com. Here's the output on my machine:

```
$ grep to: app.rb
    to: 'test-email-1@example.com',
$ ruby app.rb $API_TOKEN
You tried to send to recipient(s) that have been marked as inactive. Found inactive addresses: . Inactive recipients are ones that have generated a hard bounce, a spam complaint, or a manual suppression.
Recipients: []
```

In my production app the error that triggered this was related to a typo in the domain name, the address was something like "someone@gmial.com". The script also shows the problem for that kind of address:

```
$ grep to: app.rb
    to: 'someone@gmial.com',
$ ruby app.rb $API_TOKEN
You tried to send to recipient(s) that have been marked as inactive. Found inactive addresses: . Inactive recipients are ones that have generated a hard bounce, a spam complaint, or a manual suppression.
Recipients: []
```

**Comment by pgraham3** [comment]
@mkwiatkowski Strange I still am seeing the expected output with the provided example recipient:

```
╭─pgraham@<redacted> ~/Desktop/postmark-issue-128-poc ‹ruby-3.2.5› ‹main*› 
╰─➤  ruby app.rb
You tried to send to recipient(s) that have been marked as inactive. Found inactive addresses: test-email-1@example.com. Inactive recipients are ones that have generated a hard bounce, a spam complaint, or a manual suppression.
Recipients: ["test-email-1@example.com"]
```

In the rescue block what do you see if you output the whole error:

```
# app.rb

begin
  client.deliver(
    from: 'sender@domain.com',
    to: 'test-email-1@example.com',
    subject: 'Test Email',
    text_body: 'This is a test email'
  )
rescue Postmark::InactiveRecipientError => e
  puts e.inspect # ADD THIS
  puts e.message
  puts "Recipients: #{e.recipients.inspect}"
end
```

Adding that I see this output:

```
╭─pgraham@<redacted> ~/Desktop/postmark-issue-128-poc ‹ruby-3.2.5› ‹main*› 
╰─➤  ruby app.rb
#<Postmark::InactiveRecipientError: You tried to send to recipient(s) that have been marked as inactive. Found inactive addresses: test-email-1@example.com. Inactive recipients are ones that have generated a hard bounce, a spam complaint, or a manual suppression.>
You tried to send to recipient(s) that have been marked as inactive. Found inactive addresses: test-email-1@example.com. Inactive recipients are ones that have generated a hard bounce, a spam complaint, or a manual suppression.
Recipients: ["test-email-1@example.com"]
```

As another test - what does the raw API response body JSON look like for you if you use Postman or curl etc... to try to send to an inactive address? I get back a response body like this when interacting with the API outside of the Ruby gem:

```json
{
    "ErrorCode": 406,
    "Message": "You tried to send to recipient(s) that have been marked as inactive. Found inactive addresses: test-email-1@example.com. Inactive recipients are ones that have generated a hard bounce, a spam complaint, or a manual suppression."
}
```

**Comment by mkwiatkowski** [comment]
@pgraham3 There you go:

```sh
curl "https://api.postmarkapp.com/email" \
  -X POST \
  -H "Accept: application/json" \
  -H "Content-Type: application/json" \
  -H "X-Postmark-Server-Token: $API_TOKEN" \
  -d '{"From": "sender@domain.com", "To": "someone@gmial.com", "Subject": "Test Email", "TextBody": "This is a test email"}'
```
and the output is:

```json
{"ErrorCode":406,"Message":"You tried to send to recipient(s) that have been marked as inactive. Found inactive addresses: . Inactive recipients are ones that have generated a hard bounce, a spam complaint, or a manual suppression."}
```

At this point I wonder could it be related to the specific API token I'm using? Is it possible this bug only affects a subset of customers?

**Comment by pgraham3** [comment]
@mkwiatkowski Yeah this is super strange and could potentially be a bug with our API. Can you please generate another token for the Server (the current token will continue working - it just adds another one) and see if the fresh token has different results?

**Comment by mkwiatkowski** [comment]
@pgraham3 Tried it with two other API tokens (one within the same server as before and second in another server) and both exibit the same behavior.

**Comment by pgraham3** [comment]
Thanks for giving that a shot @mkwiatkowski!

Is there any potential for the network you are on having an impact - such as a data loss prevention mechanism stripping out email addresses? I still have not been able to reproduce a response missing the inactive email address and have not heard other reports of this happening. 

If it is possible for you, please try out the same API call on another network to see what happens. If you still see the same behavior can you reach out to us at **support@postmarkapp.com** and let them know the Server name you are using in your account? Please mention Patrick asked you to contact them so I can get notified and continue investigating.

**Comment by mkwiatkowski** [comment]
@pgraham3 Network doesn't seem to have an effect on that behavior. I see the same problem on our production app servers (hosted on Heroku) and on my local machine. I also tried it on another AWS server and no difference. I'll contact the support shortly.
